pixeltable-mcp/base-sdk/tools.py

```
@@ -232,42 +344,62 @@ def execute_query(
+        if select_columns:
+            select_args = []
+            for col_name in select_columns:
+                if not _IDENTIFIER_RE.match(col_name):
+                    return f"Error: Invalid column name: '{col_name}'"
+                if hasattr(table, col_name):
+                    select_args.append(getattr(table, col_name))
+                else:
+                    return f"Error: Column '{col_name}' not found in table '{table_name}'."
+            query = query.select(*select_args)
 
         return f"Query '{query_name}' created successfully for table '{table_name}'."
```

Ruff (0.15.1)

[warning] Consider moving this statement to an `else` block

(TRY300)